src/core/meshcop/tcat_agent.cpp

```diff
@@ -535,6 +547,86 @@ Error TcatAgent::HandlePing(const Message &aIncomingMessage,
     }
 
     SuccessOrExit(error = aOutgoingMessage.AppendBytesFromMessage(aIncomingMessage, aOffset, aLength));
+    response = true;
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetNetworkName(Message &aOutgoingMessage, bool &response)
+{
+    Error             error    = kErrorNone;
+    MeshCoP::NameData nameData = Get<MeshCoP::NetworkNameManager>().GetNetworkName().GetAsData();
+
+    VerifyOrExit(Get<ActiveDatasetManager>().IsCommissioned(), error = kErrorInvalidState);
+#if !OPENTHREAD_CONFIG_ALLOW_EMPTY_NETWORK_NAME
+    VerifyOrExit(nameData.GetLength() > 0, error = kErrorInvalidState);
+#endif
+
+    SuccessOrExit(
+        error = Tlv::AppendTlv(aOutgoingMessage, kTlvResponseWithPayload, nameData.GetBuffer(), nameData.GetLength()));
+    response = true;
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetDeviceId(Message &aOutgoingMessage, bool &response)
+{
+    uint8_t        *deviceId;
+    uint16_t        length = 0;
+    Mac::ExtAddress eui64;
+    Error           error = kErrorNone;
+
+    if (mVendorInfo->mGeneralDeviceId != nullptr)
+    {
+        length   = mVendorInfo->mGeneralDeviceId->mDeviceIdLen;
+        deviceId = const_cast<uint8_t *>(mVendorInfo->mGeneralDeviceId->mDeviceId);
+    }
+
+    if (length == 0)
+    {
+        Get<Radio>().GetIeeeEui64(eui64);
+
+        length   = sizeof(Mac::ExtAddress);
+        deviceId = eui64.m8;
+    }
+
+    SuccessOrExit(error = Tlv::AppendTlv(aOutgoingMessage, kTlvResponseWithPayload, deviceId, length));
+
+    response = true;
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetExtPanId(Message &aOutgoingMessage, bool &response)
+{
+    Error error;
+
+    VerifyOrExit(Get<ActiveDatasetManager>().IsCommissioned(), error = kErrorInvalidState);
+
+    SuccessOrExit(error = Tlv::AppendTlv(aOutgoingMessage, kTlvResponseWithPayload,
+                                         &Get<MeshCoP::ExtendedPanIdManager>().GetExtPanId(), sizeof(ExtendedPanId)));
+    response = true;
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetProvisioningUrl(Message &aOutgoingMessage, bool &response)
+{
+    Error    error = kErrorNone;
+    uint16_t length;
+
+    VerifyOrExit(mVendorInfo->mProvisioningUrl != nullptr, error = kErrorInvalidState);
+
+    length = StringLength(mVendorInfo->mProvisioningUrl, kProvisioningUrlMaxLength);
+    VerifyOrExit(length > 0 && length <= Tlv::kBaseTlvMaxLength, error = kErrorInvalidState);
+
+    error    = Tlv::AppendTlv(aOutgoingMessage, kTlvResponseWithPayload, mVendorInfo->mProvisioningUrl,
+                              static_cast<uint8_t>(length));
```

Comment:
Can remove the `static_cast<uint8_t>` now? 
(due to the change to `AppendTlv()` to accept `uint16_t`).

Reply:
fixed.